### PR TITLE
feat(medusa): Add /admin/products/:id/variants end point

### PR DESCRIPTION
**Pull request by adrien2p**
**Goal**
- Handle the retrieval of a product variants

**Improvements**
- Add a new end point such as `GET /admin/products/:id/variants`
- Update the ProductService#retrieveVariants to include the config object as usual for a list

**Tests**
- Add new unit tests for the handler
- Add new integration tests for the API

**Typings**
Introduce the typings from [there](https://github.com/medusajs/medusa/pull/1469/files#diff-f0408f39047bdd3776a42aeef2cbb7da29575bbaee2695052f5dfbae17ec91deR6)

Linked to #1461

### CODE REVIEW

File: packages/medusa/src/services/product.js [hunk 1]
```diff
@@ -388,7 +388,9 @@ class ProductService extends BaseService {
    * @return {Promise} an array of variants
    */
   async retrieveVariants(productId) {
-    const product = await this.retrieve(productId, { relations: ["variants"] })
+    const product = await this.retrieve(productId, {
+      relations: ["variants", "variants.prices"],
+    })
```
Comment (olivermrbl) [inline packages/medusa/src/services/product.js:393]: issue: Add config object containing relations, select, etc. similar to normal retrieve methods

**Comment by adrien2p** [comment]
I will do a last polish and then it will be ready

File: packages/medusa/src/api/routes/admin/products/list-variants.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    fields?.split(",") as (keyof Product)[],
+    expand
+      ? [...defaultStoreProductsVariantsRelations, ...expand.split(",")]
+      : undefined
+  )
+
+  const productService: ProductService = req.scope.resolve("productService")
+  const variants = await productService.retrieveVariants(id, {
+    ...queryConfig,
+    include_discount_prices: true,
```
[the remark this answers is not in the document]
Comment (adrien2p) [reply packages/medusa/src/api/routes/admin/products/list-variants.ts:56]: no particular reason. Should it be not part of that request?

File: packages/medusa/src/api/routes/admin/products/list-variants.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      ? [...defaultAdminProductsVariantsRelations, ...expand.split(",")]
+      : undefined
+  )
+
+  const productService: ProductService = req.scope.resolve("productService")
+  const variants = await productService.retrieveVariants(id, {
+    ...queryConfig,
+    skip: offset,
+    take: limit,
+  })
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/products/list-variants.ts:58]: thought: would it make sense to do `productVariantService.listAndCount({ product_id: id }, ...)` instead maybe?

Comment (adrien2p) [reply packages/medusa/src/api/routes/admin/products/list-variants.ts:58]: Yes completely 👌

**Review by srindom — APPROVED** [review]
LGTM!

**Comment by olivermrbl** [comment]
Please resolve conflicts @adrien2p 👍

**Comment by adrien2p** [comment]
@olivermrbl conflicts managed, let's wait the pipes to finish

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**